File: Dados C Sharp/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Dados_C_Sharp
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        int Numero = 1;
        int Vuelta = 1;

        Random random = new Random();
        private int NumeroDado()
        {
            int numero_azar = random.Next(0, 5);
            return numero_azar;
        }

        private void Cubilete_Click(object sender, EventArgs e)
        {
            CambiaDado(Dado1);
            CambiaDado(Dado2);
            CambiaDado(Dado3);
            CambiaDado(Dado4);
            CambiaDado(Dado5);

            String Turno = "Tiro número " + Numero + " de la vuelta " + Vuelta +" realizado";
            MessageBox.Show(Turno);
            Numero += 1;
            if (Numero > 3)
            {
                Numero = 1;
                Vuelta += 1;
            }
            if (Vuelta == 11)
            {
                Juego_Completado final = new Juego_Completado();
                final.ShowDialog();
            }

        }
        private void CambiaDado(PictureBox DadoX)
        {
            int numero = NumeroDado();
            switch (numero)
            {
                case 1: DadoX.Image = Properties.Resources.dado1; 
                    break;
                case 2: DadoX.Image = Properties.Resources.dado2; 
                    break;
                case 3: DadoX.Image = Properties.Resources.dado3; 
                    break;
                case 4: DadoX.Image = Properties.Resources.dado4; 
                    break;
                case 5: DadoX.Image = Properties.Resources.dado5; 
                    break;
                case 6: DadoX.Image = Properties.Resources.dado6; 
                    break;
            }
        }
    }
}

File: Dados VB/Form1.vb
Public Class Form1

    Dim Numero = 1
    Dim Vuelta = 1

    Dim Azar As Boolean = True

    'Private Sub Button1_Click(sender As Object, e As EventArgs) Handles bTirar.Click
    '    Randomize()
    '    Dim Num As Integer = Int(Rnd() * 6 + 1)
    '    MessageBox.Show(NumDado(Azar))
    'End Sub
    Function NumDado(Random As Boolean) As Integer
        If Random Then Randomize()
        Return Int(Rnd() * 6 + 1)
    End Function
    Private Sub Cubilete_Click(sender As Object, e As EventArgs) Handles Cubilete.Click
        CambiaDado(Dado1)
        CambiaDado(Dado2)
        CambiaDado(Dado3)
        CambiaDado(Dado4)
        CambiaDado(Dado5)

        Dim Turno = MsgBox("Tiro número " & Numero & " de la vuelta " & Vuelta & " realizado")
        Numero += 1
        If Numero > 3 Then
            Numero = 1
            Vuelta += 1
        End If
        If Vuelta = 11 Then
            Dim final As New Juego_Completado
            final.ShowDialog()
        End If
    End Sub
    Sub CambiaDado(ByRef DadoX As PictureBox)
        Dim Num As Integer = NumDado(Azar)
        Select Case Num
            Case 1 : DadoX.Image = My.Resources.dado1
            Case 2 : DadoX.Image = My.Resources.dado2
            Case 3 : DadoX.Image = My.Resources.dado3
            Case 4 : DadoX.Image = My.Resources.dado4
            Case 5 : DadoX.Image = My.Resources.dado5
            Case 6 : DadoX.Image = My.Resources.dado6
        End Select
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        If MessageBox.Show("¿Está seguro que desea salir del juego?", "Salir del juego", MessageBoxButtons.YesNo, MessageBoxIcon.Question, MessageBoxDefaultButton.Button2) = DialogResult.Yes Then End
    End Sub

    'Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
    '    Dim monto As Decimal = 1000
    '    Dim IVA As Decimal = 0, IIBB As Decimal = 0, interes1 As Decimal = 0
    '    CalcularValores(monto, IVA, IIBB, interes1)
    '    MessageBox.Show(" Monto: " & monto & " IVA: " & IVA & " IIBB: " & IIBB & " interes: " & interes1)
    'End Sub

    'Private Sub CalcularValores(ByVal montoX As Decimal, ByRef IVA As Decimal, ByRef IIBB As Decimal, ByRef interes1 As Decimal)
    '    IVA = montoX * 21 / 100
    '    IIBB = montoX * 3 / 100
    '    interes1 = montoX * 5 / 100
    'End Sub
End Class
